src/main/java/at/hannibal2/skyhanni/features/misc/MovementSpeedDisplay.kt

```diff
@@ -3,46 +3,43 @@ package at.hannibal2.skyhanni.features.misc
 import at.hannibal2.skyhanni.SkyHanniMod
 import at.hannibal2.skyhanni.config.enums.OutsideSbFeature
 import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
 import at.hannibal2.skyhanni.utils.LorenzUtils
 import at.hannibal2.skyhanni.utils.LorenzUtils.round
 import at.hannibal2.skyhanni.utils.LorenzVec
 import at.hannibal2.skyhanni.utils.RenderUtils.renderString
 import net.minecraft.client.Minecraft
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
-import kotlin.concurrent.fixedRateTimer
 
 class MovementSpeedDisplay {
 
     private val config get() = SkyHanniMod.feature.misc
-    private var display = ""
 
-    init {
-        fixedRateTimer(name = "skyhanni-movement-speed-display", period = 250, initialDelay = 1_000) {
-            checkSpeed()
-        }
-    }
-
-    private fun checkSpeed() {
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
         if (!isEnabled()) return
 
-        val distance = with(Minecraft.getMinecraft().thePlayer) {
+        currentBPS = with(Minecraft.getMinecraft().thePlayer) {
             val oldPos = LorenzVec(prevPosX, prevPosY, prevPosZ)
             val newPos = LorenzVec(posX, posY, posZ)
 
             // Distance from previous tick, multiplied by TPS
-            oldPos.distance(newPos) * 20
+            (oldPos.distance(newPos) * 20).round(2)
```

Comment:
this whole logic "distance from previious tick" should get moved into a utils class. maybe LocationUtils?